src/block_cache/block_chunkcache.c

```diff
@@ -556,8 +558,10 @@ __wt_chunkcache_get(WT_SESSION_IMPL *session, WT_BLOCK *block, uint32_t objectid
                 return (ret);
             }
 
-            if (__chunkcache_should_pin_chunk(session, chunk))
+            if (__chunkcache_should_pin_chunk(session, chunk)) {
                 F_SET(chunk, WT_CHUNK_PINNED);
+                WT_STAT_CONN_INCR(session, chunk_cache_chunks_pinned);
```

Comment:
I would prefer to have this in either `__chunkcache_alloc` or `__chunkcache_alloc_chunk` - it's where the existing stat is bumped, and I suspect we're eventually going to have more functions that can bring content into the cache. The tradeoff is an extra check for this flag, but perhaps the `F_SET` call could also be moved. What do you think?

Reply:
Updated in [efc862d](https://github.com/wiredtiger/wiredtiger/pull/9480/commits/efc862d902141e082a164839bb362d0ea06e5503). There is some redundancy but I don't seem to mind it, let me know what you think.